go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -94,6 +94,51 @@ var DoltScripts = []enginetest.ScriptTest{
 	},
 }
 
+// DoltUserPrivTests are tests for Dolt-specific functionality that includes privilege checking logic.
+var DoltUserPrivTests = []enginetest.UserPrivilegeTest{
+	{
+		Name: "dolt_diff table function privilege checking",
+		SetUpScript: []string{
+			"CREATE TABLE mydb.test (pk BIGINT PRIMARY KEY);",
+			"CREATE TABLE mydb.test2 (pk BIGINT PRIMARY KEY);",
+			"SELECT DOLT_COMMIT('-am', 'creating tables test and test2');",
+			"INSERT INTO mydb.test VALUES (1);",
+			"SELECT DOLT_COMMIT('-am', 'inserting into test');",
+			"CREATE USER tester@localhost;",
+		},
+		Assertions: []enginetest.UserPrivilegeTestAssertion{
+			{
+				// Without access to the database, dolt_diff should fail
+				User:        "tester",
+				Host:        "localhost",
+				Query:       "SELECT * FROM dolt_diff('test', 'main~', 'main');",
+				ExpectedErr: sql.ErrDatabaseAccessDeniedForUser,
+			},
+			{
+				// Grant access to the underlying user table
+				User:     "root",
+				Host:     "localhost",
+				Query:    "GRANT SELECT ON mydb.test TO tester@localhost;",
+				Expected: []sql.Row{{sql.NewOkResult(0)}},
+			},
+			{
+				// After granting access to `test`, dolt_diff should work
+				User:     "tester",
+				Host:     "localhost",
+				Query:    "SELECT COUNT(*) FROM dolt_diff('test', 'main~', 'main');",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				// With access to the db, but not the table, dolt_diff should fail
+				User:        "tester",
+				Host:        "localhost",
+				Query:       "SELECT * FROM dolt_diff('test2', 'main~', 'main');",
+				ExpectedErr: sql.ErrPrivilegeCheckFailed,
+			},
+		},
```

Comment:
I'd add a few more assertions. Let's revoke the `SELECT` privilege on `mydb.test`, and then grant a `SELECT` privilege on the `mydb` database and check that the function still works with the tables. Then we revoke the db-level privilege and grant a global `SELECT` privilege, just to make sure that each level works as expected.

Reply:
done deal; Thanks Daylon!